File: tests/csharp/control_flow/foreach_array.cs
// vybe-test: csharp/control_flow/foreach_array
// origin: languages/csharp/tests/csharp/test_control_flow.rs

using static __Harness;

var sum = 0;
foreach (var x in new int[] { 10, 20, 30 }) {
            sum = sum + x;
        }
__P((sum).ToString());
__Check("60");

public static class __Harness {
    public static string __buf = "";
    public static void __P(string s) { __buf = __buf + s + "\n"; }
    public static void __Pr(string s) { __buf = __buf + s; }
    public static void __Check(string want) {
        if (__buf != want && __buf != want + "\n") {
            Console.WriteLine("FAIL: want [" + want + "] got [" + __buf + "]");
            throw new Exception("assertion failed");
        }
    }
}

File: tests/vb/vb_control_flow/for_each_array.vb
' vybe-test: vb/vb_control_flow/for_each_array
' origin: languages/vb/tests/vb/test_vb_control_flow.rs

Module M
    Sub Main()
        Dim arr() As Integer = {10, 20, 30}
        Dim sum As Integer = 0
        For Each x As Integer In arr
            sum = sum + x
        Next
        Console.WriteLine(sum)
    End Sub
End Module
